### DISCUSSION TITLE
PeaNUT update failed

### DISCUSSION THREAD

**Discussion by lalandeweerd** [tteck/Proxmox#3937] [Help Needed]
Updating (with `update`) PeaNUT to v2.6.1 fails with the error:
`[ERROR] in line 68: exit code 0: while executing command pnpm run build &> /dev/null`

**Comment by tteck** [comment]
Please try again after clearing the browser cache. https://github.com/tteck/Proxmox/commit/128b076731e86dc55f5f2d9525f1dfe0edcdcbcc

**Comment by spocoeric** [comment]
Updating (or re-installing) to 3.0.0 is resulting in an Internal Server Error. The developer is saying a settings.yml is created on first launch, but that doesn't seem to be happening with the lxc

**Reply by tteck** [reply]
🏁 The script runs without any problems from this end.

```
    ____             _   ____  ________
   / __ \___  ____ _/ | / / / / /_  __/
  / /_/ / _ \/ __ `/  |/ / / / / / /
 / ____/  __/ /_/ / /|  / /_/ / / /
/_/    \___/\__,_/_/ |_/\____/ /_/

Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 111
Using Hostname: peanut
Using Disk Size: 4
Allocated Cores: 2
Allocated RAM: 2048
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Using APT-Cacher IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a PeaNUT LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using thin-pool for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 111 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.250 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.113.3
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed Node.js
 ✓ Installed NUT
 ✓ Installed Peanut
 ✓ Created Service
 ✓ Customized Container
 ✓ Cleaned
 ✓ Set Container to Normal Resources
 ✓ Completed Successfully!

PeaNUT should be reachable by going to the following URL.
         http://192.168.86.250:3000 
```
```
    ____             _   ____  ________
   / __ \___  ____ _/ | / / / / /_  __/
  / /_/ / _ \/ __ `/  |/ / / / / / /
 / ____/  __/ /_/ / /|  / /_/ / / /
/_/    \___/\__,_/_/ |_/\____/ /_/

 ✓ Updated PeaNUT to v3.0.0
root@peanut:~# 
```
Appears the issue is on your end.

Did you notice the resource warning?

**Reply by spocoeric** [reply]
Are you able to open http://192.168.86.250:3000 after the install? That's where I'm seeing the Internal Server Error

**Reply by tteck** [reply]
@remz1337 can you take a look at this?

**Comment by tteck** [comment]
**Temporally removed PeaNUT from the website.**

**Comment by Mirabis** [comment]
~New install will work but a update won't~ Script used Env variables but those are only respected on first install (breaking change) https://github.com/Brandawg93/PeaNUT/releases/tag/v3.0.0.

It tries to open the file that doesn't exist 
```
Oct 22 17:32:34 peanut peanut[2067]:   errno: -2,
Oct 22 17:32:34 peanut peanut[2067]:   code: 'ENOENT',
Oct 22 17:32:34 peanut peanut[2067]:   syscall: 'open',
Oct 22 17:32:34 peanut peanut[2067]:   path: './config/settings.yml'
Oct 22 17:32:34 peanut peanut[2067]: }
```

**Reply by tteck** [reply]
> New install will work

A new install did not work for me.

**Reply by spocoeric** [reply]
I suspect we would need a way to assign the ENV variables to the settings.yml PRIOR to the lxc container starting up. This is the contents of the settings.yml that is created when a docker image of 3.0.0 is deployed if that helps (I changed the hosts to localhost in the example).

NUT_HOST: 0.0.0.0
NUT_PORT: 3493
WEB_HOST: 0.0.0.0
WEB_PORT: 3000

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/blob/aadcb5b264f43f175ade3a26bf92f7c89b93c675/install/peanut-install.sh#L59-L63

**Comment by remz1337** [comment]
Was able to test this. PR to fix it has been submitted
https://github.com/tteck/Proxmox/pull/3953

**Comment by spocoeric** [comment]
Seems to be working now. Also, PeaNUT has updated to 3.1.0, and there is now an option to set the server host from the web interface once the internal server error is detected.

**Reply by spocoeric** [reply]
> Seems to be working now. Also, PeaNUT has updated to 3.1.0, and there is now an option to set the server host from the web interface once the internal server error is detected.

3.1.1 is now out, and can confirm that update script still not working, however fresh install is working

**Comment by DavidMoffatt-web** [comment]
I tried updating to 3.1.1 and am getting an internal server error as well. Same error as previous user when I run systemctl status peanut.

Oct 22 17:32:34 peanut peanut[2067]:   errno: -2,
Oct 22 17:32:34 peanut peanut[2067]:   code: 'ENOENT',
Oct 22 17:32:34 peanut peanut[2067]:   syscall: 'open',
Oct 22 17:32:34 peanut peanut[2067]:   path: './config/settings.yml'
Oct 22 17:32:34 peanut peanut[2067]: }

**Reply by remz1337** [reply]
Please re-create a new Peanut LXC. I'm assuming you created the LXC with the previous install script, which used to create the settings.yml file in the app folder, but it gets erased upon updating Peanut. This was fixed in the latest patch.

Alternatively, if you don't want to reinstall, you can manually create the `/etc/peanut/settings.yml` file with the following content:
```
WEB_HOST: 0.0.0.0
WEB_PORT: 3000
NUT_HOST: 0.0.0.0
NUT_PORT: 3493
```

then link it to the location where peanut is looking for it:

```
mkdir -p /opt/peanut/.next/standalone/config
ln -sf /etc/peanut/settings.yml /opt/peanut/.next/standalone/config/settings.yml
```

**Comment by spocoeric** [comment]
Confirmed able to update from 3.1.1 to 3.1.2 with update script. setting.yml either persisted, or correctly re-created. Either way, it's working. Thanks all!
